datasetinsights/estimators/builder.py

```diff
@@ -0,0 +1,101 @@
+import torch
+from tensorboardX import SummaryWriter
+
+import datasetinsights.constants as const
+from datasetinsights.estimators import Estimator
+from datasetinsights.estimators.faster_rcnn import FasterRCNNDepedencies
+from datasetinsights.storage.checkpoint import EstimatorCheckpoint
+from datasetinsights.storage.kfp_output import KubeflowPipelineWriter
+from datasetinsights.torch_distributed import init_distributed_mode, is_master
+
+
+class EstimatorBuilder:
```

Comment:
The purpose is to reuse this code used to configure the evaluate and train estimator

```
    if len(ctx.args) == 1:
        model_config.merge_from_list(ctx.args[0].split(" "))
    else:
        model_config.merge_from_list(ctx.args)

    gpu, rank, distributed = init_distributed_mode()
    if torch.cuda.is_available() and not ctx.params["no_cuda"]:
        device = torch.device("cuda")
    else:
        device = torch.device("cpu")
    logdir = ctx.params["tb_log_dir"]
    if logdir == const.NULL_STRING:
        # Use logdir=None to force using SummaryWriter default logdir,
        # which points to ./runs/<model>_<timestamp>
        logdir = None

    # todo this makes it so that we lose the tensorboard
    #  writer of non-master processes which could make debugging harder
    writer = SummaryWriter(
        logdir,
        write_to_disk=is_master(),
        max_queue=const.SUMMARY_WRITER_MAX_QUEUE,
        flush_secs=const.SUMMARY_WRITER_FLUSH_SECS,
    )
    kfp_writer = KubeflowPipelineWriter(
        filename=const.DEFAULT_KFP_METRICS_FILENAME,
        filepath=const.DEFAULT_KFP_METRICS_DIR,
    )
    checkpointer = EstimatorCheckpoint(
        estimator_name=model_config.estimator,
        log_dir=writer.logdir,
        distributed=distributed,
    )
```